command/operator_raft_join.go

```diff
@@ -143,14 +148,21 @@ func (c *OperatorRaftJoinCommand) Run(args []string) int {
 		return 2
 	}
 
-	resp, err := client.Sys().RaftJoin(&api.RaftJoinRequest{
-		LeaderAPIAddr:    leaderAPIAddr,
+	joinReq := &api.RaftJoinRequest{
 		LeaderCACert:     leaderCACert,
 		LeaderClientCert: leaderClientCert,
 		LeaderClientKey:  leaderClientKey,
 		Retry:            c.flagRetry,
 		NonVoter:         c.flagNonVoter,
-	})
+	}
+
+	if strings.HasPrefix(leaderInfo, "provider=") {
```

Comment:
Do we want provider to always be the first word in this string?

Reply:
Yes, it's the format for cloud auto-join. See [go-discover](https://github.com/hashicorp/go-discover).

Reply:
Is it though?  Roy raises a good point: there's nothing in the docs that say that the provider pair must come first in the list of kv pairs.  And the very fact that consul uses `Contains` suggests that they support it anywhere in the list.

Reply:
You're absolutely right. I thought I recalled reading the Parse logic and that this was expected.